e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
+    is not considered. This means that the base table is used (usually monthly).
+
+    Parameters
+    ----------
+    handlers_path : str
+        The path to the complex handlers, which are Python modules.
+    tables_path : str
+        The path to the CMIP6 tables.
+    var_list : List[str]
+        The list of variables to CMOR-ize.
+    freq : str, optional
+        The frequency, by default "mon".
+    realm : str, optional
+        The realm, by default "atm".
+    simple : bool, optional
+        If True, run in simple mode, by default False
+
+    Returns
+    -------
+    List[Dict[str, str]]:
+        A list of dictionaries, with each dictionary storing information related
+        to the CMOR handler of each variable.
     """
-    given the base table (for monthly data), find the correct
-    table given the frequency. Check the tables directory that this table
-    exists and the variable is included
+    available_handlers = _get_available_handlers(handlers_path)
+    chosen_handlers: List[Dict[str, Any]] = []
+
+    if "all" in var_list:
+        for key, handler in available_handlers.items():
+            # Update the handler's "table" if the frequency is not monthly.
+            if not simple and freq != "mon":
+                available_handlers[key]["table"] = _get_table_for_non_monthly_freq(
+                    handler["name"], handler["table"], freq, realm, tables_path
+                )
+
+        print_message(
+            f"Loaded all available handlers with a frequency of {freq}.",
+            "debug",
+        )
+        chosen_handlers = [handler for handler in available_handlers.values()]
+
+        return chosen_handlers
+
+    for var in var_list:
+        # Use the high frequency version of a variable handler based on the
+        # specified frequency. For example, for "pr" with a frequency of "day"
+        # or "3hr, use "pr_highfreq.py" instead of "pr.py".
+        # TODO: Once the handler modules are refactored using classes, we can
+        # update this check to something more sophisticated.
+        if _use_highfreq_handler(var, freq):
+            var += "_highfreq"
+
+        handler = available_handlers.get(var)
+        if handler is None:
+            raise KeyError(
+                f"Variable '{var}' failed to load. Either an entry is not defined in "
+                "the default handlers file, or a Python module of the same name does "
+                "not exist."
+            )
+
+        # Update the handler's "table" if the frequency is not monthly.
+        if not simple and freq != "mon":
+            handler["table"] = _get_table_for_non_monthly_freq(
+                handler["name"], handler["table"], freq, realm, tables_path
+            )
+
+        chosen_handlers.append(handler)
+        print_message(
+            f"Loaded `{var}` handler for variable `{handler['name']}`.", "debug"
+        )
+
+    return chosen_handlers
```

Comment:
This section handles if the user passes specific handlers to the list of variables.